crates/engine/tree/src/tree/mod.rs

```diff
@@ -613,6 +613,7 @@ where
         loop {
             match self.try_recv_engine_message() {
                 Ok(Some(msg)) => {
+                    debug!(target: "engine", "received new engine message");
```

Comment:
maybe log something from the engine message itself, like a message type?

Reply:
this will also include the full block which is very noisy, but should be fine for debug